exonum-java-binding/core/src/main/proto/service.proto

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package exonum.binding;
+
+option java_package = "com.exonum.binding.common.messages";
+
+/**
+TODO: move this file to exonum-proto-sources-repo
+currently this action is blocked till native update to the latest revision
+*/
+
+// An object for the service configuration which passed to `service#initialize` method.
+// It brings a possibility to configure a service using any of provided formats.
+// The format provides the type of the configuration encoded as string value.
+message ServiceConfiguration {
+    enum Format {
+        NONE = 0;
```

Comment:
What is the purpose of NONE? If it is neither TEXT nor any from the other formats,    then the messages shall not be used.

Reply:
the purpose of None that protobuf may instantiate a message from bytes with default values in fields.
Actually It happens in test:
```
  @Test
  void getAsPlainStringIsNotServiceConfiguration() {
    byte[] serializedConfig = anyId().toByteArray();
    ServiceConfiguration serviceConfiguration = new ServiceConfiguration(serializedConfig);

    assertThrows(IllegalArgumentException.class, serviceConfiguration::getAsString);
  }
```
the underlying object has format=None and value = ""

Reply:
If the format is unset, it will [have the 0 enum value](https://developers.google.com/protocol-buffers/docs/proto3#enum), which we can define to be plain text, which is acceptable. 
None/plain_text as a default for unset doesn't have much difference: you can always get it as string, but cannot decode in any other format.

> the underlying object has format=None and value = ""

This particular test does not test what it says: it takes some array that _may_ be decoded, and verifies there is _some_ error. It must pass something that protobuf cannot decode correctly, and verify it rejects it. Here, it seems, it just accepts the byte array as filled with defaults (zeroes).